src/scaffoldmaker/meshtypes/meshtype_1d_path1.py

```diff
@@ -229,7 +232,22 @@ def extractPathParametersFromRegion(region, groupName=None):
         cd13.append(d13)
         node = nodeIter.next()
 
-    return cx, cd1, cd2, cd12, cd3, cd13
+    result = ()
+    for label in valueLabels:
+        if label == Node.VALUE_LABEL_VALUE:
+            result += (cx,)
+        if label == Node.VALUE_LABEL_D_DS1:
+            result += (cd1,)
+        if label == Node.VALUE_LABEL_D_DS2:
+            result += (cd2,)
+        if label == Node.VALUE_LABEL_D_DS3:
+            result += (cd3,)
+        if label == Node.VALUE_LABEL_D2_DS1DS2:
+            result += (cd12,)
+        if label == Node.VALUE_LABEL_D2_DS1DS3:
+            result += (cd13,)
```

Comment:
This isn't the solution I'm looking for; should instead build the result directly and loop over the number of valueLabels and call getNodeParameters only for those requested in the code earlier.

Reply:
I modified the code so it only gets the values specified by valueLabels. What we return depends on the values specified by the valueLabels, I do not know if there is another way to do this. Can you please tell me what else can be done?

Reply:
Combine with the evaluation loop:
```
valueLabelsCount = len(valueLabels)
returnValues = ([] for i in range(valueLabelsCount)
while node.isValid():
        cache.setNode(node)
        for i in range(valueLabelsCount):
            result, values  = coordinates.getNodeParameters(cache, -1, valueLabels[i], 1, componentsCount)
            for c in range(componentsCount, 3):
                values.append(0.0)
            returnValues[i].append(values)
        node = nodeIter.next()
return returnValues
```

Reply:
Brilliant! Thanks.